src/Dafda/Diagnostics/DafdaActivitySource.cs

```diff
@@ -96,8 +100,8 @@ public static Activity StartPublishingActivity(PayloadDescriptor carrier)
     public static Activity StartPublishingActivity(OutboxEntry entry, string clientId)
     {
         PropagationContext parentContext = default;
-        var messageType = string.Empty;
```

Comment:
not a big fan of `object` and casting it later. It makes the serialization hack leak more than it has to... how about this?:

```csharp       
        if (!TryDeserializePayload(entry.Payload, out var payload))
        {
            return null;
        }

        // extract message type and id
        payload.TryGetValue("type", out var messageTypeToken);
        payload.TryGetValue("messageId", out var messageIdToken);
        var messageType = messageTypeToken?.ToString();
        var messageId = messageIdToken?.ToString();

        // Extract the context injected in the metadata by the publisher
        var parentContext = Propagator.Extract(default, payload, ExtractContextFromDictionary);

        // Inject extracted info into current context
        Baggage.Current = parentContext.Baggage;
        ```

Reply:
Ok I will try it like this and make the team test it as well.
 I tested it on the component that broke down because of the serialisation issue before.